### PR TITLE
#457 - Update schema to allow for WBS element deletion

### LINKED ISSUE #457: Admin Tools - Schema Changes for WP and Project Deletion
### Description

we are going to do a soft delete like all our other stuff so we need to update the schema to allow for that. we also need to modify anywhere this might be a problem in this ticket

### Acceptance Criteria

update the schema in the most efficient way to add this. This might just mean adding a deleted field to WBS Element, but it could take a lot more. I'm purposefully not giving a lot of detail because this is a very advanced ticket and you should be able to figure it out yourself (mostly, definitely consult with me!)

### Proposed Solution

update the schema and create a new migration

brainstorm places where we need to filter out deleted work packages and projects (all  GET endpoints for example)

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by jonathan-chen10**
## Changes

Modify `User` to keep track of deleted WBS elements and CRs. Modify `Wbs_Element` and `Change_Request` schemas to include deletion time and user.

Edit test data accordingly.

## Notes

I looked through the risk log schema ticket (Northeastern-Electric-Racing/PM-Dashboard-v2#698) and copied much of the deletion information.

## To Do

These still need to be done before merging.

- [x] Migrate schema officially
- [x] Fix tests again

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #457

### CODE REVIEW

File: src/backend/src/utils/work-packages.utils.ts [hunk 1]
```diff
@@ -25,6 +25,9 @@ export const getWbsElementId = async ({
       wbsNumber: { carNumber, projectNumber, workPackageNumber }
     }
   });
+
+  if (wbsElem?.dateDeleted) return undefined;
```
Comment (jamescd18) [inline src/backend/src/utils/work-packages.utils.ts:29]: I don't know much about this function. Does returning undefined match the expected behavior by consumer code? Does returning undefined also match the behavior of prisma.findUnique in the case where the WBS element is not found? just wanna be sure.

File: src/backend/src/services/change-request.services.ts [hunk 2]
```diff
@@ -332,8 +336,8 @@ export default class ChangeRequestsService {
       }
     });
 
-    if (wbsElement === null)
-      throw new NotFoundException('WBS Element', `${carNumber}.${projectNumber}.${workPackageNumber}`);
+    if (!wbsElement) throw new NotFoundException('WBS Element', `${carNumber}.${projectNumber}.${workPackageNumber}`);
```
[the remark this answers is not in the document]
Comment (anthonybernardi) [reply src/backend/src/services/change-request.services.ts:339]: hmm i can move that to shared i guess

File: src/backend/src/utils/projects.utils.ts [hunk 3]
```diff
@@ -42,11 +42,16 @@ export const manyRelationArgs = Prisma.validator<Prisma.ProjectArgs>()({
       }
     },
     team: true,
-    goals: true,
-    features: true,
-    risks: riskQueryArgs,
-    otherConstraints: true,
+    goals: { where: { dateDeleted: null } },
```
[the remark this answers is not in the document]
Comment (anthonybernardi) [reply src/backend/src/utils/projects.utils.ts:45]: yeah it takes the logic tot he backend which i like

File: src/backend/src/utils/work-packages.utils.ts (hunk 1 above)
Comment (anthonybernardi) [reply src/backend/src/utils/work-packages.utils.ts:29]: i dont like this function at all but this is how it was before so i just went with it

Comment (anthonybernardi) [reply src/backend/src/utils/work-packages.utils.ts:29]: actually it's only used in one spot so im gonna change it cuz it sucks

**Comment by anthonybernardi** [comment]
the tests i deleted were ones that were related to the user not being found and stuff which can't happen anymore since we're using the auth way to get the user before even entering the service function

**Review by anthonybernardi — APPROVED** [review]
nice

**Pull request merged**